pandas/tests/indexes/multi/test_setops.py

```diff
@@ -414,6 +414,14 @@ def test_union_empty_self_different_names():
     tm.assert_index_equal(result, expected)
 
 
+def test_union_multiindex_empty_rangeindex():
+    # GH#41234
+    mi = MultiIndex.from_arrays([[1, 2], [3, 4]], names=["a", "b"])
+    ri = pd.RangeIndex(0)
+    tm.assert_index_equal(mi, mi.union(ri), check_names=False)
```

Comment:
this is wrong, it should be of type Index or we should raise, this shouldn't just silently work

cc @jbrockmendel

Reply:
hmm we actually return the non-empty even when these are no the same type

Reply:
Not quite sure what you mean with

> it should be of type Index or we should raise

This should copy the behavior to 1.2.4.

Reply:
i think this is fine.  its values-dependent which we're trying to move away from, but it is consistent with what we do in all the other Index set methods

Reply:
yep